### ISSUE TITLE
USSD on MTK HELIO A22 Xiaomi redmi 6A 

### ISSUE THREAD

**Issue by sergeyadams2002** [phhusson/treble_experimentations#426]
ussd doen not work on Xiaomi Redmi 6A MTK helio a22 platform: mt6765

**Comment by phhusson** [comment]
try to do setprop persist.sys.radio.ussd.fix true

**Comment by artur-g** [comment]
Same issue on Vernee T3 Pro MT6739WA AOSP 8.1 v29 system-arm64-aonly-gapps-su.img.xz 
Su made by Magisk 18.1.
Tested in a terminal emulator with su rights. Getprop persist.sys.radio.ussd.fix before applying setprop, returned an empty value.
After reboot value is still true, USSD does not work (it seems to be sending but not receiving).

**Comment by oscx** [comment]
@artur-g do You observe this with both SIM slots, or different behavior? With Oukitel C11 Pro, which also has MT6739 and dual SIM, i have different behavior SIM slot1/2, one OK, one limited to 2G/EDGE. 

logcat logs generating a lot of lines like 
> E ProxyController: ERROR: sending down the same id for different phones

**Comment by artur-g** [comment]
I had MTK devices for a long time (Lolipop and up) and what I observed from my adventures with them, is that ONLY ONE slot is capable of 4g or 3g. Another one is always limited. 

That said, you can always choose which one is faster. My current setup: Sim1 - calls and sms, Sim 2 - 4g internet (no call or sms capable as far as I know). On Sim2 I get around 2-3 megabytes/sec speed (21.1Mbps where I sit right now) and 4g icon is showing (dont know if it is lying or not but speed is ok)
I cant test simultanious internet access because I have internet on Sim 1 turned off on carrier level.

**Comment by oscx** [comment]
@artur-g yes, thanks for your reply and info. only with stock rom, i get both slots to be able to connect 4G (not at the same time, but each single one). however, not a big problem, not critical in my opinion.. anyway, confused by the repeating error from logs.

**Comment by artur-g** [comment]
@oscx I tested swapping sim cards.
Sim1 - Internet
Sim2 - Phone
USSD still does not work. But I was also unable to force Sim1 into 4g mode. Tried limiting sim2 into 2g, manually registering into network, reboots, nothing worked.
Conclusion: Mediatek stock system apps may control this stuff.
Solution: Invade China and force them to give us the code :)

**Comment by oscx** [comment]
good luck :)

**Comment by phhusson** [comment]
can you detail what type of USSD?

also please provide adb logcat -b all both on stock and on GSI when doing that ussd

**Comment by artur-g** [comment]
@phhusson 
USSD is *108# to check, how much money do I have on a prepaid account (Polish operator "Heyah"). It normally works by dialing this code in dialler, pushing "dial" and after that, I get (on both roms) "USSD code running...". I only don't get the response message on GSI (with amount of money left on account). 

I will provide logs on weekend. I need time to backup and reset system to stock. Also I'm currently using GSI with Xposed Framework and other stuff so logs from that can be dirty and give you couple false positives. Expect logs on weekend.
 
Thank you for taking the time and trying to help with that. Also, great work. Best rom I ever had.

**Comment by svoboda18** [comment]
USSD will never work in aosp 8.1, as the fix isn't implemented (Yet). If you want be sure it is still a bug, test with latest aosp 9.0.

**Comment by artur-g** [comment]
Sorry for late responce.
Vernee T3 Pro MT6739WA AOSP 9.0 v111 system-arm64-aonly-gapps-su.img.xz
getprop persist.sys.radio.ussd.fix after first reboot returns true
USSD works out of the box (and is very fast).
I don't think that logs are needed, just fix needs to be implemented in 8.1 release.
Thanks for the support and you guys are doing GREAT work.

**Comment by dwapking** [comment]
Using Oukitel K8 here.
Vendor = stock Oreo 8.0
GSI = AOSP v111, Pie.
SoC = MT6750.

I had the USSD issue also, able to dial any code but no reply (such as when checking account balance).

The issue was however resolved with the build.prop tweak, adding persist.sys.radio.ussd.fix=true

Now it works fine even after countless reboots.
Thanks @phhusson

**Comment by phhusson** [comment]
Do you have one of those files?

/vendor/lib/mtk-ril.so /vendor/lib64/mtk-ril.so /vendor/lib/libmtk-ril.so /vendor/lib64/libmtk-ril.so

Do you have magisk installed?

**Comment by artur-g** [comment]
I only have this  /vendor/lib64/libmtk-ril.so (I know the question wasn't addressed to me, but more info more better)
Magisk above v18 does not work on 9.0 GSI for me. It's probably "early mount" bug on Magisk end https://github.com/topjohnwu/Magisk/issues/844
